plugins/aws/fix_plugin_aws/resource/ssm.py

```diff
@@ -351,45 +353,111 @@ class AwsSSMNonCompliantSummary:
+        if spec := cls.api_spec:
+            try:
+                kwargs = spec.parameter or {}
+                items = builder.client.list(
+                    aws_service=spec.service,
+                    action=spec.api_action,
+                    result_name=spec.result_property,
+                    **kwargs,
+                )
+                cls.collect(items, builder)
+            except Boto3Error as e:
+                msg = f"Error while collecting {cls.__name__} in region {builder.region.name}: {e}"
+                builder.core_feedback.error(msg, log)
+                raise
+            except Exception as e:
+                msg = f"Error while collecting {cls.__name__} in region {builder.region.name}: {e}"
+                builder.core_feedback.info(msg, log)
+                raise
+
+    def parse_finding(self) -> Finding:
+        title = self.title or ""
+        severity = SEVERITY_MAPPING.get(self.severity or "", Severity.medium)
+        details = self.compliance_details
+        if self.execution_summary:
+            updated_at = self.execution_summary.execution_time
+        else:
+            updated_at = None
+        return Finding(title, severity, None, None, updated_at, details)
+
+    @classmethod
+    def collect(cls, json: List[Json], builder: GraphBuilder) -> None:
+        def add_finding(
+            provider: str, finding: Finding, clazz: Optional[Type[AwsResource]] = None, **node: Any
+        ) -> None:
+            if resource := builder.node(clazz=clazz, **node):
+                resource.add_finding(provider, finding)
+
+        def collect_compliance_items(jsons: List[Json]) -> None:
+            spec = AwsApiSpec("ssm", "list-compliance-items", "ComplianceItems")
+            compliance_ids = [item["ResourceId"] for item in jsons]
+            for result in builder.client.list(
+                aws_service="ssm",
+                action=spec.api_action,
+                result_name=spec.result_property,
+                expected_errors=spec.expected_errors,
+                ResourceIds=compliance_ids,
+            ):
+                if finding := AwsSSMResourceCompliance.from_api(result, builder):
+                    if (
+                        (rt := finding.resource_type)
+                        and (rid := finding.resource_id)
+                        and (clazz := ResourceTypeLookup.get(rt))
+                    ):
+                        # append the finding when all resources have been collected
+                        builder.after_collect_actions.append(
+                            partial(
+                                add_finding,
+                                "amazon_ssm_compliance",
+                                finding.parse_finding(),
+                                clazz,
+                                id=rid,
+                            )
+                        )
+
+        # we can request only 40 items per request
+        for jsons in chunks(json, 39):
```

Comment:
The documentation says this:
> Currently, you can only specify one resource ID per call. List results depend on the criteria specified in the filter.

Reply:
I tested the limit and found that it scales to 40 items rather than one